x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts

```diff
@@ -281,6 +285,7 @@ export function buildComponentTemplates(params: {
     (dynampingTemplate) => Object.keys(dynampingTemplate)[0]
   );
 
+  // In case of index_mode: time_series this should be set to true by default except it is overwritten
```

Comment:
Makes sense to me - we should try to map out the relationship between synthetic source + time series indexing in code to make this clear. Rather have some declarative structure that describes that relationship rather than a bunch of chained conditionals.

Reply:
++. There should be a single place to look up and change all this logic. Similar to a state machine.